sympy/utilities/lambdify.py

```diff
@@ -934,16 +934,18 @@ def _cse_del(expr):
     ([(x0, x + cos(x)), (x1, x0*sin(x0))], [set(), {x0}], sqrt(x1) + x1)
     """
     def get_free_symbols(expr):
-        if hasattr(expr, "__iter__") and not isinstance(expr, str):
-            return set.union(*map(get_free_symbols, expr))
         try:
             return expr.free_symbols
         except AttributeError:
             return set()
+        if hasattr(expr, "__iter__") and not isinstance(expr, str):
```

Comment:
if you pass an iterable it will return set() because of the try/except block above

Reply:
Also, we typically use `iterable` to do this test:
```python
>>> from sympy.core.compatibility import iterable
>>> iterable('asd')
False
>>> iterable([])
True
```

Reply:
> if you pass an iterable it will return set() because of the try/except block above

My God! The big beginner's mistake! We will say that I was really not in good shape that day ... Thank you for showing it to me;)

Reply:
> Also, we typically use `iterable` to do this test:
> 
> ```python
> >>> from sympy.core.compatibility import iterable
> >>> iterable('asd')
> False
> >>> iterable([])
> True
> ```

just like 'cse', i didn't know about the existence of this function. I just integrated it, it's true that it's a little more elegant.